src/components/Post/RandomButton.test.tsx

```diff
@@ -0,0 +1,75 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import axios from 'axios';
+import { useRouter } from 'next/router';
+
+import RandomButton, { postRandomButtonTestIds } from './RandomButton';
+
+jest.mock('axios');
+jest.mock('next/router', () => ({
+  __esModule: true,
+  useRouter: jest.fn(),
+}));
+
+describe('RandomButton', () => {
+  it('renders the button', () => {
+    render(<RandomButton />);
+
+    const randomButton = screen.getByTestId(postRandomButtonTestIds.root);
+
+    expect(randomButton).toBeVisible();
+  });
+
+  describe('when clicked', () => {
+    const mockPush = jest.fn();
+
+    beforeEach(() => {
+      (useRouter as jest.Mock).mockImplementation(() => ({
+        push: mockPush,
+      }));
+    });
+
+    describe('and the returned slug is NOT empty', () => {
+      it('redirects to a random post', async () => {
+        const response = {
+          data: {
+            slug: 'example',
+            excludedSlugs: ['example'],
+          },
+        };
+        const mockedAxios = axios as jest.Mocked<typeof axios>;
+        mockedAxios.post.mockResolvedValueOnce(response);
+
+        render(<RandomButton />);
+
+        const randomButton = screen.getByTestId(postRandomButtonTestIds.root);
+        await randomButton.click();
```

Comment:
Could you tell me the reason for using `await randomButton.click();` over `userEvent.click(randomButton)` and `fireEvent.click(randomButton)`? 😢

Reply:
I didn't add `@testing-library/user-event` before so I used `await randomButton.click();`.
But we should use that package to simulate the event instead. Added